cartographer/mapping/internal/2d/scan_matching/ceres_scan_matcher_2d.cc

```diff
@@ -70,12 +71,22 @@ void CeresScanMatcher2D::Match(const Eigen::Vector2d& target_translation,
                                    initial_pose_estimate.rotation().angle()};
   ceres::Problem problem;
   CHECK_GT(options_.occupied_space_weight(), 0.);
-  problem.AddResidualBlock(
-      CreateOccupiedSpaceCostFunction2D(
-          options_.occupied_space_weight() /
-              std::sqrt(static_cast<double>(point_cloud.size())),
-          point_cloud, grid),
-      nullptr /* loss function */, ceres_pose_estimate);
+  CHECK(grid.grid_type() != GridType::NONE);
```

Comment:
As discussed, get rid of GridType::None.
1. make Grid2D::grid_type() const pure virtual
2. drop Grid2D::grid_type_, return const values instead